Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -37,6 +38,46 @@ internal class DefaultRepository<TItem> : IRepository<TItem> where TItem : IItem
             ILogger<DefaultRepository<TItem>> logger) =>
             (_optionsMonitor, _containerProvider, _logger) = (optionsMonitor, containerProvider, logger);
 
+        public async ValueTask<IPage<TItem>> PageAsync(Expression<Func<TItem, bool>> predicate, CancellationToken cancellationToken = default, int pageSize = 25,
+            int page = 1, string continuationToken = null)
+        {
+            if (page < 1) throw new ArgumentOutOfRangeException(nameof(page));
+            if (pageSize < 1) throw new ArgumentOutOfRangeException(nameof(pageSize));
+
+
+            Container container = await _containerProvider.GetContainerAsync().ConfigureAwait(false);
+            Expression<Func<TItem, bool>> queryPredicate = predicate.Compose(item => !item.Type.IsDefined() || item.Type == typeof(TItem).Name, Expression.AndAlso);
+            IQueryable<TItem> query = null;
+            QueryRequestOptions queryOptions = new() { MaxItemCount = pageSize };
+
+            if (continuationToken is null)

```

Comment:
Don't see any difference in the `continuationToken` branch. How about simplifying:

```csharp
query = container.GetItemLinqQueryable<TItem>(false, continuationToken, queryOptions).Where(queryPredicate);
if (page is not 1)
{
    query = query.Skip((page * pageSize) - pageSize);
}
```